tests/test_expectation_maximization.py

```diff
@@ -99,11 +104,63 @@ def test_generate_slices(test_ir, n_particles, n_pix):
     rots = test_ir.grid_SO3_uniform(n_particles)
     xy_plane = test_ir.generate_xy_plane(n_pix)
 
-    slices, xyz_rotated = test_ir.generate_slices(map_3d, xy_plane, n_pix, rots)
+    slices, xyz_rotated_planes = test_ir.generate_slices(map_3d, xy_plane, n_pix, rots)
 
-    assert xy_plane.shape == (n_pix**2, 3)
     assert slices.shape == (n_particles, n_pix, n_pix)
-    assert xyz_rotated.shape == (n_pix**2, 3)
+    assert xyz_rotated_planes.shape == (n_particles, 3, n_pix**2)
+
+    # ---------------------------------
+    # 90-degree rotation test.
+    # Map has ones in central xz-plane.
+    # Rotating by -90 degrees about y
+    # should produce a slice of ones.
+    rot_test_map = np.zeros((4, 4, 4))
+    rot_test_map[2] = np.ones((4, 4))
+
+    rot_mat = np.array(
+        [
+            [[0, 0, -1], [0, 1, 0], [1, 0, 0]],
+        ]
+    )
+
+    xy_plane = np.array(
+        [
+            [-2, -1, 0, 1, -2, -1, 0, 1, -2, -1, 0, 1, -2, -1, 0, 1],
+            [-2, -2, -2, -2, -1, -1, -1, -1, 0, 0, 0, 0, 1, 1, 1, 1],
+            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        ]
+    )
+
+    slices, xyz_rotated_planes = test_ir.generate_slices(
+        rot_test_map, xy_plane, 3, rot_mat
```

Comment:
n_pix is 3 here. should be 4.

better to hard code it once
n_pix = 4, and replace where its needed.